### ISSUE TITLE
Proposal: Use URL field to store IPFS Hash of review content

### ISSUE THREAD

**Issue by atfornes** [bloxberg-org/peer-review-app#29]
When adding a review to the blockchain, the URL field could store a IPFS address of the review content the review does not have a public URL. That way, the content of the review can be openly shared and not only stored in the database.

**Comment by crossoveranx** [comment]
In theory IPFS could be used for most of the storage on the application. In the past, I've experienced quite a few issues regarding performance from IPFS - so I'm hesitant to integrate this. Additionally, a centralized database enables us to move certain information to a central DB when we do a more thorough review of the data in order to be GDPR compliant. I don't believe we can provide these guarantees if we use IPFS as another entity could be pin the data. What are your thoughts?

**Comment by atfornes** [comment]
At Bloxberg summit I heard that as long as somebody knows they are publicly publishing something, (e.g. an academic article) the right to erase such information from GDPR regulation does not apply. It would be great to have that expert (I don't remember his name) giving his opinion on this topic and confirming I did not understand it wrong.

 Also, I think there is an analogy with traditional web publishing that applies: if a website publishes something, other people can copy and share that information after the website removes it. The same applies to IPFS. We could promote ways to ask third parties to unpin the data.

It is an interesting debate. Our current position is that if a reviewer wants to openly share the content of a review, that content should be available to all the network, and not controller and kept by a single entity.

**Comment by atfornes** [comment]
I found the slide where Jörn Erbguth shared the strategies to be GDPR compliant:

![GDPR](https://user-images.githubusercontent.com/7475584/84896811-e3ba8b80-b0a4-11ea-9139-327347393a84.png)

**Comment by crossoveranx** [comment]
> It is an interesting debate. Our current position is that if a reviewer wants to openly share the content of a review, that content should be available to all the network, and not controller and kept by a single entity.

I agree with this, I also don't think it should be owned by a single entity. At this stage of the project, it would likely be a lot of overhead to implement, and I would like to focus on other features. However, I would happily accept a PR as long as it's clear that the researchers have the option to choose whether to put it on IPFS or a private storage, it should be fine. I do wonder about the performance of IPFS in most instances, last time I experimented with it (2~ years ago), it was quite poor
